pinot-common/src/main/java/org/apache/pinot/common/utils/RoundRobinURIProvider.java

```diff
@@ -54,6 +55,14 @@ public RoundRobinURIProvider(URI originalUri)
     _index = new Random().nextInt(_uris.length);
   }
 
+  public RoundRobinURIProvider(List<URI> uris) {
```

Comment:
javadocs say that this class is supposed to resolve the hostname to multiple IP addresses. This ctor breaks that definition.

Reply:
Was able to re-purpose most of the things here and the naming is close to what we exactly want! Maybe we can modify javadocs accordingly. 
cc @Jackie-Jiang wdyt?

Reply:
+1 on also resolve the hostname to multiple IPs to follow the existing behavior. We can probably add a helper method to resolve a given URI to a list of URIs

Reply:
Updated. Please check.